### PR TITLE
[AIRFLOW-5600] Add MLEngine system tests

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5600
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [flake8]
 max-line-length = 110
-ignore = E731,W504,I001
+ignore = E731,W504,I001,I004
```
Comment (potiuk) [inline .flake8]: As discussed - we should not add it.

**Review by potiuk — APPROVED** [review]

File: airflow/gcp/example_dags/example_mlengine.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    )
+
+    list_version_result = BashOperator(
+        bash_command="echo \"{{ task_instance.xcom_pull('list-version') }}\"",
+        task_id="list-version-result",
+    )
+
+    prediction = MLEngineBatchPredictionOperator(
+        task_id="prediction",
+        project_id="polidea-airflow",
```
Comment (yk5) [inline airflow/gcp/example_dags/example_mlengine.py:150]: PROJECT_ID ?

File: airflow/gcp/example_dags/example_mlengine.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    }
+}
+
+with models.DAG(
+    "example_gcp_mlengine", default_args=default_args, schedule_interval=None  # Override to match your needs
+) as dag:
+    training = MLEngineTrainingOperator(
+        task_id="training",
+        project_id=PROJECT_ID,
+        region="us-central1",
```
Comment (yk5) [inline airflow/gcp/example_dags/example_mlengine.py:63]: It would be better to make this as an env variable, too.

Comment (mik-laj) [reply airflow/gcp/example_dags/example_mlengine.py:63]: Recently, we have been trying to limit the creation of environment variables to a minimum.  We create them only to provide isolation between different tests. Any user can run these tests in this region, so no parameterization is needed here

File: airflow/gcp/example_dags/example_mlengine.py (hunk 2 above)
Comment (mik-laj) [reply airflow/gcp/example_dags/example_mlengine.py:150]: Thanks. I am currently creating PR to improve it.

Comment (mik-laj) [reply airflow/gcp/example_dags/example_mlengine.py:150]: https://github.com/apache/airflow/pull/6698

**Pull request merged**